### ISSUE TITLE
Migrate FE Tools-owned services from CircleCI to GitHub Actions 

### ISSUE THREAD

**Issue by meganhkelley** [department-of-veterans-affairs/va.gov-team#21718]

**Comment by meganhkelley** [comment]
[2/23/21 announcement](https://dsva.slack.com/archives/CJRQ85PQB/p1614124642018200)

Jesse House:no_entry:  6:57 PM
Hello All,
We wanted to share some important information regarding Continuous Integration (CI) on VSP. This information has been reviewed and approved by @Dror Matalon, @Michael Fleet, and @Demian Ginther.
Because of a number of limitations and the administrative burden associated with CircleCI, the Operations team will be pursuing GitHub actions as its primary CI solution. Operations will strongly recommend, support, and provide examples and best practices to other teams for CI pipelines using GitHub actions. However, the vision is that Operations Team does not wish to require that other teams can only use GitHub actions, as it is envisioned that Ops can be agnostic when it comes to the choice of a specific CI tool, as long as that tool safely integrates with platform infrastructure.
High-level background:
The problem with CircleCI for the Operations team's use case is that there are credentials to manage at the repository level across many different projects, and managing them all as separate pieces of code is tedious, error prone, and makes CircleCI a bad fit given the current limitations that the VA has in place for the administration of software repositories.
Additional information:
There is an ADR being drafted regarding this decision, and additional feedback is welcome: https://github.com/department-of-veterans-affairs/va.gov-team/pull/20042
The summary of the ADR thus far is:
"CircleCI as implemented with VA restrictions does not allow easy management of permissions on 'Contexts' which contain credentials which are available to the entire 'department-of-veterans-affairs' Github Org. Although they are not displayed in plaintext in the CircleCI UI, they are obtainable in plaintext by running a build in CircleCI and using SSH to connect to your in-progress build, and outputting ENV variables. If the entire Org has access to the context, there is no way to stop anyone in the Org from doing so.
In addition, CircleCI as implemented with VA restrictions does not allow use of 'third-party' Orbs (reusable code) which means that we cannot publish our own Orbs either.
These two items are not preventing the use of CircleCI completely, as it is still possible to manage per-project secrets (only available to those with Github permissions on that repo) and you are free to cut and paste code from one workflow to another to re-use code, however, it makes managing multiple repos that have similar functions very tedious, and error prone."
More info:
While a specific CI workflow will not be required, there will be a required workflow for application deployments, currently implemented using ArgoCD. The Operations team will document what ArgoCD wants to receive as input, and how application teams can produce that input. Additionally, application teams will be able to log into ArgoCD to view, access logs and events, and troubleshoot their application deployments.
TL;DR
Ops is choosing a new tool for its internal CI workflow (GitHub actions).
This tool will not be required for other teams, but will be strongly recommended and fully supported.
Teams can choose a different CI tool if it better fits their workflow and expertise.
There will be a required workflow for CD (ArgoCD).
Ops will provide documentation, examples, and best practices for CI/CD pipelines, and will assist other teams in setting them up, based on the above.
This is the current plan for CI, and we look forward to feedback so that we can understand your team's needs and ensure that we do what's best for the platform.
Thank you! :pray: (edited)
